multiverse/approvalweight_manager.go

```diff
@@ -38,14 +42,14 @@ func (a *ApprovalManager) ApproveMessages(messageID MessageID) {
 	weight := a.tangle.WeightDistribution.Weight(issuingMessage.Issuer)
 	a.tangle.Utils.WalkMessagesAndMetadata(func(message *Message, messageMetadata *MessageMetadata, walker *walker.Walker) {
 
-		weightByte := message.WeightSlice[int(byteIndex)]
-		if weightByte&(1<<mod) != 0 {
+		weightByte := messageMetadata.weightSlice[int(byteIndex)]
+		if weightByte&(1<<mod) == 0 {
 			weightByte |= 1 << mod
-			message.Weight += weight
-			a.Events.MessageWeightUpdated.Trigger(message.ID)
-			if float64(message.Weight) >= config.MessageWeightThreshold*float64(a.tangle.WeightDistribution.TotalWeight()) {
-				message.ConfirmationTime = time.Now()
-				a.Events.MessageConfirmed.Trigger(a.tangle.Peer.ID, message.Issuer, message.ID, message.IssuanceTime, message.ConfirmationTime, message.Weight)
+			messageMetadata.weight += weight
```

Comment:
We will need to update the `messageMetadata.weightSlice` by adding 
`messageMetadata.weightSlice[int(byteIndex)] = weightByte`

Reply:
( The way of preventing data copy is addresses in newer comment)

Reply:
I think it should be `[]*byte` because you want to create an array of pointer and not a pointer to an array

anyhow when I tried to do that I got an error. I can probably make it work but I prefer to not preoptimize until we hit an issue.